server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -520,6 +518,14 @@ public static FieldSortBuilder getPrimaryFieldSortOrNull(SearchSourceBuilder sou
         return null;
     }
 
+    @SuppressWarnings("unchecked")
+    private static <T extends Comparable<T>> MinAndMax<T> extractMinAndMax(IndexReader reader, String fieldName,
+                                                              Function<byte[], Comparable<?>> converter) throws IOException {
```

Comment:
should it be a `Function<byte[], ? extends T>` to avoid the unchecked cast?

Reply:
great point, I missed this